dotCMS/src/main/java/com/dotcms/api/system/event/Payload.java

```diff
@@ -54,29 +65,41 @@ public Payload() {
 	 */
 	public Payload(final Visibility visibility,
 				   final String visibilityId) {
+		this(new Void(), visibility, visibilityId, null);
+	}
 
-        this(new Void(), visibility, visibilityId);
-    }
-
-    /**
-     * Creates a payload object.
-     *
-     * @param data         {@link Object}
-     *                     - Any Java object that represents the payload.
-     * @param visibility   {@link Visibility}
-     *                     - If the event should be apply just for a specific user, role or global
-     * @param visibilityId {@link String}
-     *                     - Depending of the visibility type, this could be an userId or roleId, for global just keep it null.
-     */
-    public Payload(final Object data,
-                   final Visibility visibility,
-                   final String visibilityId) {
+	public Payload(final Object data,
+				   final Visibility visibility,
+				   final String visibilityId) {
+		this(data, visibility, visibilityId, null);
+	}
 
-        this.type = data.getClass().getName();
-        this.data = data;
-        this.visibility = visibility;
-        this.visibilityId = visibilityId;
-    }
+	/**
+	 * Creates a payload object.
+	 * 
+	 * @param data {@link Object}
+	 *            - Any Java object that represents the payload.
+	 * @param visibility {@link Visibility}
+	 * 			  - If the event should be apply just for a specific user, role or global
+	 * @param visibilityId {@link String}
+	 * 			  - Depending of the visibility type, this could be an userId or roleId, for global just keep it null.
+	 */
+	public Payload(final Object data,
+				   final Visibility visibility,
+				   final String visibilityId,
+				   final Map<String, Object> user) {
+
+		this.type = data.getClass().getName();
+		this.data = data;
+		this.visibility = visibility;
+		this.visibilityId = visibilityId;
+
+		if (user == null || user.isEmpty()) {
```

Comment:
Also it might have issues in terms of testing, please try to keep the Bean's cleaner